include/pagmo/problems/lennard_jones.hpp

```diff
@@ -0,0 +1,182 @@
+
+This file is part of the PaGMO library.
+
+The PaGMO library is free software; you can redistribute it and/or modify
+it under the terms of either:
+
+  * the GNU Lesser General Public License as published by the Free
+    Software Foundation; either version 3 of the License, or (at your
+    option) any later version.
+
+or
+
+  * the GNU General Public License as published by the Free Software
+    Foundation; either version 3 of the License, or (at your option) any
+    later version.
+
+or both in parallel, as here.
+
+The PaGMO library is distributed in the hope that it will be useful, but
+WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received copies of the GNU General Public License and the
+GNU Lesser General Public License along with the PaGMO library.  If not,
+see https://www.gnu.org/licenses/. */
+
+#ifndef PAGMO_PROBLEM_LENNARDJONES_HPP
+#define PAGMO_PROBLEM_LENNARDJONES_HPP
+
+#include <iostream>
+#include <limits>
+#include <stdexcept>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <pagmo/exceptions.hpp>
+#include <pagmo/problem.hpp> // needed for cereal registration macro
+#include <pagmo/types.hpp>
+
+namespace pagmo
+{
+
+/// The Lennard Jones Cluster problem.
+/**
+ *
+ * \image html lennard_jones.jpg "Pictures of Lennard-Jones clusters." width=3cm
+ *
+ * This is a box-constrained continuous single-objecive problem. It represents the minimization
+ * of the energy of a cluster of atoms assuming a Lennard-Jones potential between each pair.
+ * The complexity for computing the objective function scales with the square of the number of atoms.
+ *
+ * The decision vector contains [z2, y3, z3, x4, y4, z4, ....] as the cartesian coordinates x1, y1, z1, x2, y2 and x3
+ * are fixed to be zero.
+ *
+ * See: http://doye.chem.ox.ac.uk/jon/structures/LJ.html
+ *
+ */
+class lennard_jones
+{
+public:
+    /// Constructor from number of atoms
+    /**
+     * Constructs a Lennard Jones Clusters global optimisation problem
+     *
+     * @param atoms number of atoms in the cluster.
+     *
+     * @throw std::invalid_argument if \p atoms is < 3
+     */
+    lennard_jones(unsigned atoms = 3u) : m_atoms(atoms)
+    {
+        if (atoms < 3) {
+            pagmo_throw(std::invalid_argument, "The number of atoms in a Lennard Jones Clusters problem must be "
+                                               "positive and greater than 2, while a number of "
+                                                   + std::to_string(atoms) + " was detected.");
+        }
+    }
+    /// Fitness computation
+    /**
+     * Computes the fitness for this UDP
+     *
+     * @param x the decision vector.
+     *
+     * @return the fitness of \p x.
+     */
+    vector_double fitness(const vector_double &x) const
+    {
+        vector_double f(1, 0.);
+        double sixth, dist;
```

Comment:
Unless I am mistaken, there is no need to declare these variables outside the for loop. Just create them on the fly within the loop, so it's clear these are not values that need to survive past the loop.